### DISCUSSION TITLE
Menu configuration

### DISCUSSION THREAD

**Discussion by GaryElshaw** [losnoco/Cog#315] [Ideas]
The current default intimates that the slider is for the volume rather than the current track. I think it would be worth moving the slider with either a new identity or perhaps moving it elsewhere in the bar.

![SCR-20220719-tp2](https://user-images.githubusercontent.com/69029666/179718777-48492d69-49dc-4d1f-afde-f23e8af5ef70.png)
.

**Comment by kode54** [comment]
How about I just hide the volume control instead?

**Reply by GaryElshaw** [reply]
I'm not mean, or trivial, Chris. Cog is my favourite music player, but every now and again i confuse those two controls because of their proximity. Entirely possible i'm alone, in which case, ignore me.

**Comment by kode54** [comment]
How about if I default to adding an extra spacer between the seekbar and the volume button, would that help? You can add a spacer yourself in the mean time.

**Reply by GaryElshaw** [reply]
It's not about me, really. I'd just move the default volume control away from the seekbar to avoid any confusion. Part of my thinking is about the separation of controls that are vertical and horizontal.

**Comment by kode54** [comment]
I didn't intend to be mean, sorry. I'm just not good at UI design, and this design has been pretty much the same forever. Also, I can't merge the seekbar and the time display into a single box, as they're separate items.

**Comment by kode54** [comment]
You can change what it is for now by ctrl or right clicking on the toolbar and using the customize option. I can change the default to be clearer, though.

**Reply by GaryElshaw** [reply]
I think it would be good to change the default proximity. I'd love to talk with you at a later date about UI. I think Cog could benefit by looking at the default horizontal and vertical controls. I really appreciate your work. Cog is my default music application.

**Comment by kode54** [comment]
Is this an adequate spacer? I can't really tell
<img width="373" alt="Screen Shot 2022-07-19 at 6 11 51 AM" src="https://user-images.githubusercontent.com/796316/179759001-093dcbc8-2e80-4a6f-b12a-0bd0e3d8dcc2.png">
.

**Reply by GaryElshaw** [reply]
The other side. It's making a change to differentiate the scrobbler from the volume control.

**Reply by GaryElshaw** [reply]
I am not a fan, but because the UI can be complex, i think it's worth considering how Cog prioritises controls. An easy, but massively uncomfortable fix is to introduce bars between the ui elements in the way that title, artist, album are, using some form of bifurcation like the current table is. Title | Artist | Album. I think it's incredibly inelegant, and think Cog deserves better.

**Reply by GaryElshaw** [reply]
I think it's worth looking at the immediate controls. By that i mean that volume should follow 'next' and the scrobbler following. They are the immediate controls.

**Reply by kode54** [reply]
Hmm, I'll see what I can do.

**Comment by kode54** [comment]
I've tried swapping the order. This should be clearer, at least. Incidentally, macOS deprecated separators in toolbars, hence why you can't see them in the customize palette, even though the default tool bar has at least one separator inserted. Nope, all you get is reordering things, and inserting either fixed or flexible spacers.

<img width="373" alt="Screen Shot 2022-07-19 at 6 24 39 AM" src="https://user-images.githubusercontent.com/796316/179761815-a76e1c4f-3359-4eea-9d2a-a794abcf8124.png">

Oh, and for this screen shot, I've enabled Increase Contrast. I've been told that Flat UI, a thing Apple introduced with Sierra, kind of makes it hard to see what is actually clickable in a UI now. I tend to agree somewhat. Also, occasionally, turning on Increase Contrast feels like a breath of fresh air, until I decide I prefer the translucency feature more.

**Reply by GaryElshaw** [reply]
Ha! Yes, I have increased contrast (outside of the accessibility controls) on by default. My ageing eyes appreciate. And funnily enough i prefer dark mode as i age.

This is how i think the default should appear.

![SCR-20220720-285](https://user-images.githubusercontent.com/69029666/179763936-e0e6dfb9-a2a7-4b96-8348-8defdcdd860e.png)

**Reply by kode54** [reply]
Gotcha, putting that together now.

**Comment by kode54** [comment]
Here's something entirely different.
![Screen Shot 2022-07-19 at 6 33 47 AM](https://user-images.githubusercontent.com/796316/179763779-83948ae3-fc74-42a1-a0c7-37b498dce623.png)

**Reply by kode54** [reply]
Yes, I have some weird music in my player.

**Reply by GaryElshaw** [reply]
No shame. We all do.

**Reply by GaryElshaw** [reply]
I'm sorry i haven't been able to isolate why the year|genre and No|Rating are stuck together how they are.

**Reply by kode54** [reply]
Rating isn't even supposed to be on that side, I put it on the left by the Index.

**Reply by GaryElshaw** [reply]
I've looked multiple times now, and i know it's something simple i'm overlooking, but i can't see it.

**Comment by kode54** [comment]
How's this attempt?
![Screen Shot 2022-07-19 at 7 14 13 AM](https://user-images.githubusercontent.com/796316/179772425-65e2887e-cb51-49a0-8a7a-01d4517350ee.png)

**Reply by GaryElshaw** [reply]
That looks great, Chris! I'll hum along to Zelda.

**Comment by GaryElshaw** [comment]
I don't know if you remember, but what feels like an age ago, i asked how i might make the FontSizetoLineHeightTransformer code work with Transmission? There's been a lot of work since then and we finally have autolayout in place. I'm wondering about next steps. I've managed to get the .mm and header files to compile fine in cmake, but was wondering if you had time to talk me through what to do next? Here's my fork https://github.com/GaryElshaw/transmission.git and the branch is 'font-size' if you want to have a look at where i've got to.
